### PR TITLE
Support changing color of the Git action part of the prompt

### PR DESCRIPTION
**Pull request by ashb**
I do a lot of rebasing and cherry-picking and I like to know when I'm in that mode, so I added an option to make it stand out more.

A picture is worth a thousand words:

Screenshot 2019-07-17 at 11 03 54

### CODE REVIEW

**Comment by ashb** [comment]
@mafredri @sindresorhus Could you take a look please? Small PR to change the color of the "you are in the middle of a rebase" marker.

File: pure.zsh [hunk 1]
```diff
@@ -248,18 +252,19 @@ prompt_pure_async_vcs_info() {
 	# to be used or configured as the user pleases.
 	zstyle ':vcs_info:*' enable git
 	zstyle ':vcs_info:*' use-simple true
-	# Only export two message variables from `vcs_info`.
-	zstyle ':vcs_info:*' max-exports 2
-	# Export branch (%b) and Git toplevel (%R).
+	# only export three msg variables from vcs_info
```
Comment (sindresorhus) [inline pure.zsh]: Don't do unrelated changes.

**Comment by sindresorhus** [comment]
Can you also add it to the ASCII diagram in https://github.com/sindresorhus/pure#colors?

File: pure.zsh (hunk 1 above)
Comment (ashb) [reply pure.zsh]: Done. Not sure how I did that, possibly started the PR from an older version of the code (I still changed "two" to "three")

File: pure.zsh [hunk 2]
```diff
@@ -134,7 +134,11 @@ prompt_pure_preprompt_render() {
 	# Add Git branch and dirty status info.
 	typeset -gA prompt_pure_vcs_info
 	if [[ -n $prompt_pure_vcs_info[branch] ]]; then
-		preprompt_parts+=("%F{$git_color}"'${prompt_pure_vcs_info[branch]}${prompt_pure_git_dirty}%f')
+		local branch="%F{$git_color}"'${prompt_pure_vcs_info[branch]}'
+		if [[ -n $prompt_pure_vcs_info[action] ]]; then
+			branch+="|%F{$prompt_pure_colors[git:action]}"'$prompt_pure_vcs_info[action]'"%F{$git_color}"
+		fi
+		preprompt_parts+=("$branch"'${prompt_pure_git_dirty}%f')
```
[the remark this answers is not in the document]
Comment (sindresorhus) [reply pure.zsh]: As mentioned in https://github.com/sindresorhus/pure/pull/491#pullrequestreview-298831312, I don't think this change is necessary.

File: pure.zsh [hunk 3]
```diff
@@ -134,7 +134,11 @@ prompt_pure_preprompt_render() {
 	# Add Git branch and dirty status info.
 	typeset -gA prompt_pure_vcs_info
 	if [[ -n $prompt_pure_vcs_info[branch] ]]; then
-		preprompt_parts+=("%F{$git_color}"'${prompt_pure_vcs_info[branch]}${prompt_pure_git_dirty}%f')
+		local branch="%F{$git_color}"'${prompt_pure_vcs_info[branch]}'
+		if [[ -n $prompt_pure_vcs_info[action] ]]; then
+			branch+="|%F{$prompt_pure_colors[git:action]}"'$prompt_pure_vcs_info[action]'"%F{$git_color}"
```
[the remark this answers is not in the document]
Comment (ashb) [reply pure.zsh]: Given it was used on the next line in a format that does end with `%f` I didn't think it necessary here.

File: readme.md [hunk 4]
```diff
@@ -97,15 +98,15 @@ The following diagram shows where each color is applied on the prompt:
...
-│          │       │        ┌───────────────── host
-│          │       │        │
-~/dev/pure master* ⇡ zaphod@heartofgold 42s
-venv ❯               │                  │
-│    │               │                  └───── execution_time
-│    │               └──────────────────────── user
-│    └──────────────────────────────────────── prompt
-└───────────────────────────────────────────── virtualenv
+│          │      ┌─────────────────────────── git:action
+│          │      │         ┌───────────────── host
```
[the remark this answers is not in the document]
Comment (ashb) [reply readme.md]: Oh yes, not sure how I managed that. Corrected in a fixup commit. Let me know if you'd like me to squash this down.

**Comment by sindresorhus** [comment]
Nice work :)
